### PR TITLE
Fix untar permissions and error handling

### PR DESCRIPTION
**Pull request by kmeaw**

### CODE REVIEW

File: untar.go [hunk 1]
```diff
@@ -208,6 +208,7 @@ func UnTarIndex(ctx context.Context, dst string, index Index, s Store, n int, op
 		req      = make(chan requestJob)
 		assemble = make(chan chan []byte, n)
 	)
+	ctx, cancel := context.WithCancel(ctx)
```
Comment (folbricht) [inline untar.go]: Going to need a `defer cancel()` here to avoid a context leak.

File: untar.go [hunk 2]
```diff
@@ -284,6 +294,11 @@ func UnTarIndex(ctx context.Context, dst string, index Index, s Store, n int, op
 	// Run untar in the main goroutine
 	err := UnTar(ctx, r, dst, opts)
 
+	if err != nil {
```
Comment (folbricht) [inline untar.go]: With the `defer cancel()` in place above, all of this (to the end of the function) could be written as:
```go
	// Run untar in the main goroutine
	if err := UnTar(ctx, r, dst, opts); err != nil {
		return err
	}
	return g.Wait()
```

But thinking a bit more about this, what do you think about running the UnTar() inside a goroutine as well with g.Go() like the others? Then we wouldn't need the extra context, and there's no question about which error gets priority. All would be covered by `return g.Wait()` at the end. That should make things a little simpler.

Comment (kmeaw) [reply untar.go]: Thank you for such a nice idea; done.

**Comment by folbricht** [comment]
Thanks for those fixes.

**Pull request merged**